x/leveragelp/keeper/msg_server_close.go

```diff
@@ -11,34 +14,43 @@ import (
 func (k msgServer) Close(goCtx context.Context, msg *types.MsgClose) (*types.MsgCloseResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	return k.Keeper.Close(ctx, msg)
-}
-
-func (k Keeper) Close(ctx sdk.Context, msg *types.MsgClose) (*types.MsgCloseResponse, error) {
-	closedPosition, repayAmount, err := k.CloseLong(ctx, msg)
+	creator := sdk.MustAccAddressFromBech32(msg.Creator)
+	position, err := k.GetPosition(ctx, creator, msg.Id)
 	if err != nil {
 		return nil, err
 	}
 
-	if k.hooks != nil {
-		ammPool, err := k.GetAmmPool(ctx, closedPosition.AmmPoolId)
-		if err != nil {
-			return nil, err
-		}
-		err = k.hooks.AfterLeverageLpPositionClose(ctx, sdk.MustAccAddressFromBech32(msg.Creator), ammPool)
-		if err != nil {
-			return nil, err
-		}
```

Comment:
just to confirm we are calling this hook later, right ?

Reply:
Yes, earlier it was missing for begin block closing so I put it in CheckHealthStopLossThenRepayAndClose